db-ops/create-db-user.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# docker image with postgres client only
+DOCKER_IMAGE_TAG=governmentpaas/psql:latest
+
+DB_ENDPOINT=$(aws rds describe-db-instances --region=$REGION --query "DBInstances[?DBName=='$PROJECT_NAME'].Endpoint.Address" | jq -r '.[0]')
+SECRET_ID=$(aws secretsmanager list-secrets --region $REGION  --query "SecretList[?Name=='$PROJECT_NAME-$ENVIRONMENT-rds-$SEED'].Name" | jq -r ".[0]")
+# RDS MASTER
+MASTER_RDS_USERNAME=master_user
+SECRET_PASSWORD=$(aws secretsmanager get-secret-value --region=$REGION --secret-id=$SECRET_ID | jq -r ".SecretString")
+# APPLICATION DB ADMIN
+DB_APP_USERNAME=$PROJECT_NAME
+DB_APP_PASSWORD=$(openssl rand -base64 16)
+
+# Fill in env-vars to db user creation manifest
+eval "echo \"$(cat ./db-ops/job-create-db.yml.tpl)\"" > ./k8s-job-create-db.yml
+# the manifest creates 4 things
+# 1. Namespace: db-ops 
+# 2. Secret in db-ops: db-create-users (with master password, and a .sql file
+# 3. Job in db-ops: db-create-users (runs the .sql file against the RDS given master_password from env)
+# 4. Secret in Application namespace with DB_USERNAME / DB_PASSWORD 
+kubectl create -f ./k8s-job-create-db.yml
+
+# Deleting the entire db-ops namespace, leaving ONLY application-namespace's secret behind
+kubectl wait --for=condition=complete --timeout=10s job db-create-users
+kubectl delete namespace db-ops
```

Comment:
If the intent is for the previous command to kill the script if it fails you should use `set -e` at the top but it's probably a better ideal to check if the return value of the previous command is zero and then delete the namespace. We probably want to keep the namespace if the job failed to debug.

Reply:
somelike like?

```
EXIT_CODE=$?
if [ $EXIT_CODE -eq 0 ]
then 
  kubectl delete namespace db-ops
if
```

Reply:
Yeah! (Though needs syntax fix, and you could just have $? in the comparison without assigning it.) And you could have an else in there with a nicer message like "Failed to create